File: microphone.cpp
/*
uses a microphone sensor to detect the ambient sound levels and light up an LED according to the detected sound levels:

*/


// Define the analog pin to which the microphone sensor is connected
#define MIC_PIN A0

// Define the digital pin to which the LED is connected
#define LED_PIN 13

// Define the threshold value for the sound levels
#define THRESHOLD 500

void setup() {
  // Set the LED pin as output
  pinMode(LED_PIN, OUTPUT);
  // Start the serial communication
  Serial.begin(9600);
}

void loop() {
  // Read the analog value from the microphone sensor
  int micValue = analogRead(MIC_PIN);
  // Print the analog value to the serial monitor
  Serial.println(micValue);
  // If the sound level is above the threshold, turn on the LED
  if (micValue > THRESHOLD) {
    digitalWrite(LED_PIN, HIGH);
  } else {
    digitalWrite(LED_PIN, LOW);
  }
  // Wait for a short amount of time
  delay(10);
}

File: microphone.py
#uses a microphone sensor to detect the ambient sound levels and light up an LED according to the detected sound levels:

import serial
import pygame

# Initialize Pygame
pygame.init()
# Set the screen size
screen = pygame.display.set_mode((640, 480))

# Open the serial port
ser = serial.Serial('COM3', 9600)

# Run the game loop
while True:
    # Read the serial data
    data = ser.readline().strip().decode('utf-8')
    # Parse the data as an integer
    try:
        value = int(data)
    except ValueError:
        continue
    # Calculate the brightness of the screen based on the sound level
    brightness = max(min(value, 1023), 0) / 1023
    # Set the screen color
    screen.fill((int(255 * brightness), int(255 * brightness), int(255 * brightness)))
    # Update the screen
    pygame.display.flip()
